pallets/portfolio/src/lib.rs

```diff
@@ -0,0 +1,262 @@
+#![cfg_attr(not(feature = "std"), no_std)]
+
+use frame_support::{
+    decl_error, decl_event, decl_module, decl_storage, dispatch::DispatchResult, ensure,
+    StorageDoubleMap,
+};
+use frame_system::{self as system, ensure_signed};
+use polymesh_common_utilities::{identity::Trait as IdentityTrait, CommonTrait, Context};
+use polymesh_primitives::{IdentityId, PortfolioId, PortfolioName, PortfolioNumber, Ticker};
+use sp_arithmetic::traits::Saturating;
+use sp_std::prelude::Vec;
+
+type Identity<T> = pallet_identity::Module<T>;
+
+pub trait Trait: CommonTrait + IdentityTrait {
+    type Event: From<Event<Self>> + Into<<Self as frame_system::Trait>::Event>;
+}
+
+decl_storage! {
+    trait Store for Module<T: Trait> as Session {
+        /// The set of existing portfolios with their names. If a certain pair of a DID and
+        /// portfolio number maps to `None` then such a portfolio doesn't exist. Conversely, if a
+        /// pair maps to `Some((num, name))` then such a portfolio exists and is called `name`.
+        ///
+        /// The portfolio number is both the key and part of the value due to a limitation of the
+        /// iterator on `StorageDoubleMap`.
+        pub Portfolios get(fn portfolios):
+            double_map hasher(blake2_128_concat) IdentityId, hasher(twox_64_concat) PortfolioNumber =>
+            Option<(PortfolioNumber, PortfolioName)>;
```

Comment:
since we can now walk over keys of a double map, it's not required to store the `PortfolioNumber` in the value as well. In fact, I don't see the need for storing Portfolio name onchain.

Reply:
I'm also in favour of removing names from the module completely. Need a confirmation from the product team.